shiny/experimental/ui/_accordion.py

```diff
@@ -0,0 +1,427 @@
+
+# @describeIn accordion_panel_set close [accordion_panel()]s.
+# @export
+def accordion_panel_close(
+    id: str,
+    values: bool | str | list[str],
+    session: Optional[Session] = None,
+) -> None:
+    _accordion_panel_action(id=id, method="close", values=values, session=session)
+
+
+# @param panel an [accordion_panel()].
+# @param target The `value` of an existing panel to insert next to. If
+#   removing: the `value` of the [accordion_panel()] to remove.
+# @param position Should `panel` be added before or after the target? When
+#   `target` is `NULL` (the default), `"after"` will append after the last
+#   panel and `"before"` will prepend before the first panel.
+#
+# @describeIn accordion_panel_set insert a new [accordion_panel()]
+# @export
+def accordion_panel_insert(
+    id: str,
+    panel: AccordionPanel,
+    target: Optional[str] = None,
+    position: Literal["after", "before"] = "after",
+    session: Optional[Session] = None,
+) -> None:
+    if position not in ("after", "before"):
+        raise ValueError("`position` must be either 'after' or 'before'")
+    session = require_active_session(session)
+    send_panel_message(
+        id,
+        session,
+        method="insert",
+        panel=session._process_ui(panel.resolve()),
+        target=None if target is None else assert_str(target),
+        position=position,
+    )
+
+
+# @describeIn accordion_panel_set remove [accordion_panel()]s.
+# @export
+def accordion_panel_remove(
+    id: str,
+    target: str | list[str],
+    session: Optional[Session] = None,
+) -> None:
+    if not isinstance(target, list):
+        target = [target]
+
+    send_panel_message(
+        id,
+        session,
+        method="remove",
+        target=assert_list_str(target),
+    )
+
+
+# @describeIn accordion_panel_set update a [accordion_panel()].
+# @inheritParams accordion_panel
+# @export
+def accordion_panel_update(
+    id: str,
+    target: str,
+    *body: TagChild,
+    title: TagChild | None | MISSING_TYPE = MISSING,
+    value: str | None | MISSING_TYPE = MISSING,
+    icon: TagChild | None | MISSING_TYPE = MISSING,
+    session: Optional[Session] = None,
+) -> None:
+    session = require_active_session(session)
+    T = TypeVar("T")
+
+    def missing_none_x(x: T | None | MISSING_TYPE) -> T | Literal[""] | None:
+        if isinstance(x, MISSING_TYPE):
+            return None
+        if x is None:
+            return ""
+        return x
+
+    title = missing_none_x(title)
+    value = missing_none_x(value)
+    icon = missing_none_x(icon)
+    send_panel_message(
+        id,
+        session,
+        method="update",
+        target=assert_str(target),
+        value=None if value is None else assert_str(value),
+        body=None if len(body) == 0 else session._process_ui(body),
+        title=None if title is None else session._process_ui(title),
+        icon=None if icon is None else session._process_ui(icon),
+    )
+
+
+def assert_str(x: str) -> str:
```

Comment:
Add `__all__` to this file (to _not_ export this function).

Reply:
(In this file...) All methods not exported now start with `_`